source/agora/common/Block.d

```diff
@@ -132,8 +149,39 @@ public struct Block
     public void serialize (scope SerializeDg dg) pure const nothrow @safe
     {
         serializePart(this.header, dg);
+
+        serializePart(this.txs.length, dg);
         foreach (ref tx; this.txs)
             serializePart(tx, dg);
+
+        serializePart(this.merkle_tree.length, dg);
+        foreach (ref merkle; this.merkle_tree)
+            dg(merkle[]);
```

Comment:
@TrustHenry : This diff is fixing serialization, not deserialization. Please split it into atomic commits.
Let's do this together when you have a bit of time.